### PR TITLE
Foundation for i18n System

### PR DESCRIPTION
**Pull request by couimet**
Establishes the foundation for internationalization (i18n) in the VSCode extension, enabling future localization to additional languages (French, Spanish, etc.) without touching UI call sites.

## Overview

This PR implements issue #104 through 6 comprehensive steps, introducing a centralized message management system with stateful locale support. Currently English-only, the foundation is ready for multi-language expansion.

**Architecture:** Extension-only i18n (core library stays English for technical messages)

## Step 1: Move MessageCode Enum to Extension Layer

**Breaking change:** MessageCode enum migrated from `rangelink-core-ts` to `rangelink-vscode-extension`

**Rationale:**

- User-facing messages belong in UI layer (extension), not core library
- Core library messages stay English (industry standard - like TypeScript, Git, JavaScript)
- Clear separation: technical errors (core) vs user-facing messages (extension)

**Changes:**

- Moved enum from `packages/rangelink-core-ts/src/types/RangeLinkMessageCode.ts` to `packages/rangelink-vscode-extension/src/types/MessageCode.ts`
- Used `git mv` to preserve file history
- Removed core export from `rangelink-core-ts/src/types/index.ts`

## Step 2: Implement Stateful Locale Management

**Components created:**

1. **`supportedLocales.ts`** - Single source of truth for all supported locales
   - `LocaleCode` type: `'en'` (expandable to `'en' | 'fr' | 'es'`)
   - `MessageMap` type: Maps MessageCode → localized string
   - `DEFAULT_LOCALE` constant

2. **`LocaleManager.ts`** - Stateful locale management (mirrors LogManager pattern)
   - `setLocale(locale: string)` - Auto-extracts base locale (`'en-US'` → `'en'`)
   - `getMessages()` - Returns current locale's message map
   - `getCurrentLocale()` - Returns active LocaleCode

3. **`messages.en.ts`** - English message templates
   - 6 message templates with `{paramName}` placeholder syntax
   - Alphabetically sorted for maintainability

**Initialization:** `extension.ts` calls `setLocale(vscode.env.language)` at startup

**Design decisions:**

- Option C hybrid: Store locale string, resolve message map on access (no stale references)
- Locale string extraction encapsulated in `setLocale()` (callers use raw `vscode.env.language`)
- Graceful fallback: Missing translations warn + fallback to English

## Step 3: Create formatMessage Infrastructure Tests

**Test approach:** Library/NPM module pattern - completely decoupled from UI messages

**Infrastructure (27 tests):**

- `TestMessageCode` enum (test-only, not real UI MessageCode)
- `testMessagesEn` map (test fixtures, not messages.en.ts)
- Runtime injection: `supportedLocales.en = testMessagesEn`
- Result: UI message changes won't break infrastructure tests

**Coverage:**

- Basic formatting: Simple messages, empty params, no placeholders
- Parameter substitution: String/numeric params, multiple params, missing params
- Complex types: Arrays, objects, circular refs (fallback to String())
- Error handling: Unknown codes throw RangeLinkExtensionError
- Locale management: setLocale, override param, region codes, fallback
- Edge cases: undefined/zero/negative/float parameters

**Custom Jest matcher:** `toThrowRangeLinkError` for clean error validation

**Test coverage:** 93%+ for formatMessage.ts

## Step 4: Complete MessageCode Migration + Logger Decoupling

**Cleanup phase:** Remove old MessageCode usage from config layer

**Changes:**

- Removed `RangeLinkMessageCode` imports from `config/logging.ts` and `config/loadDelimiterConfig.ts`
- Removed `code: MessageCode.*` parameters from logger calls (3 total)
- Updated comments in core layer to reflect new architecture

**Rationale:**

- MessageCode is for UI i18n messages, not internal logger context
- Logger context should use semantic keys (fn, field, source), not message codes
- Reduces coupling between config layer and i18n system

## Step 5: Migrate UI Messages to formatMessage()

**4 call sites migrated:**

1. **RangeLinkService.ts:132** - Status bar with parameter interpolation

   ```typescript
   // Before: `✓ ${linkTypeName} copied to clipboard`
   // After: formatMessage(MessageCode.STATUS_BAR_LINK_COPIED_TO_CLIPBOARD, { linkTypeName })
   ```

2. **ClaudeCodeDestination.ts:127** - Claude Code paste instructions

   ```typescript
   formatMessage(MessageCode.INFO_CLAUDE_CODE_LINK_COPIED);
   ```

3. **CursorAIDestination.ts:172** - Cursor AI paste instructions

   ```typescript
   formatMessage(MessageCode.INFO_CURSOR_AI_LINK_COPIED);
   ```

4. **extension.ts:125** - Commit hash copied message
   ```typescript
   formatMessage(MessageCode.INFO_COMMIT_HASH_COPIED);
   ```

**Test fix:** Added `language: 'en'` to vscode.env mock in extension.test.ts

## Step 6: Add Integration Tests

**Test approach:** Minimal automated integration + real messages.en.ts (not mocks)

**7 integration tests added:**

- **RangeLinkService (3 tests):** formatMessage called with STATUS_BAR_LINK_COPIED_TO_CLIPBOARD
  - Validates parameter interpolation (`{linkTypeName}`)
  - Tests "RangeLink" and "Portable RangeLink" parameters
- **CursorAIDestination (2 tests):** formatMessage called with INFO_CURSOR_AI_LINK_COPIED
  - Validates spy interception and message content
- **extension.ts (2 tests):** formatMessage called with INFO_COMMIT_HASH_COPIED
  - Tests command registration and message display

**Integration test characteristics:**

- Use real messages.en.ts (intentional coupling - feature, not bug)
- Spy on formatMessage() to validate MessageCode parameter
- Assert final message content matches expected from messages.en.ts
- Co-located with unit tests in existing test files

**Rationale:**

- Infrastructure tests use mocks for decoupling (formatMessage.test.ts)
- Integration tests should use real messages to catch regressions
- Minimal tests (2-3 per component) verify wiring without re-testing internals

## Architecture Benefits

- ✅ **Centralized messages:** Single source of truth (messages.en.ts)
- ✅ **Future-ready:** Add French/Spanish by creating messages.fr.ts/messages.es.ts
- ✅ **Type-safe:** MessageCode enum prevents typos
- ✅ **Stateful:** Runtime locale switching supported
- ✅ **Testable:** 93%+ infrastructure coverage, decoupled tests
- ✅ **Graceful fallback:** Missing translations warn + use English
- ✅ **Consistent:** All UI messages use same formatMessage() system

## Testing

- ✅ All 513 tests passing (23 test suites, +7 integration tests)
- ✅ TypeScript compiles without errors
- ✅ 81.79% statement coverage, 75.47% branch coverage
- ✅ New test files: formatMessage.test.ts (27 tests), localeCompleteness.test.ts (3 tests)

## Future Expansion

**To add a new language (e.g., French):**

1. Add `'fr'` to `LocaleCode` type in `supportedLocales.ts`
2. Create `messages.fr.ts` with French translations
3. Add to `supportedLocales` map: `fr: messagesFr`
4. Run `localeCompleteness.test.ts` to validate all keys present
5. No changes needed to UI call sites (already using formatMessage())

### CODE REVIEW

File: docs/ROADMAP.md [hunk 1]
```diff
@@ -328,7 +328,7 @@ Integrate terminal link provider with Phase 4A.2 config change detection. Rebuil
 
 ---
 
-## Phase 10: Internationalization (i18n)
+## Phase 10: Internationalization (i18n) - IN PROGRESS ([#37](https://github.com/couimet/rangeLink/issues/37))
```
Comment (couimet) [inline docs/ROADMAP.md]:
```suggestion
## Phase 10: Internationalization (i18n) - (See [#37](https://github.com/couimet/rangeLink/issues/37))
```
